### PR TITLE
Rewrite Polyhash interface

### PR DESCRIPTION
**Pull request by wailo**
- Add toCoordinates, toLocationIds
- API calls are symmetric.
- It easier to convert from compressed cluster to list of locationIds

### CODE REVIEW

File: src/polyhash/index.js [hunk 1]
```diff
@@ -79,39 +79,122 @@ const decode = [
...
+      const locationIs = decompress(input)
+      result = toCoordinates(locationIs)      
+    }
+    // polyhash
+    return result;
+  } catch (err) {
+    console.error(
+      `Failed to convert ${JSON.stringify(
+        input
+      )} to coordinates, reason: ${err}`
```
Comment (jake-unl) [inline src/polyhash/index.js:156]: I think it's probably best to hide some errors from our users. We can catch them and return our own error messages for some possible cases

File: src/polyhash/index.js [hunk 2]
```diff
@@ -285,8 +368,9 @@ function _toTurfPolygon(inputPolygon) {
         polygons.push(turfHelpers.polygon(inputPolygon.geometry.coordinates))
     }
     else {
+      const coords = toCoordinates(inputPolygon)
       // Polygon is an array of points
-        polygons.push(turfHelpers.polygon([inputPolygon]));
+        polygons.push(turfHelpers.polygon([coords]));
```
Comment (jake-unl) [inline src/polyhash/index.js:373]: minor: indentation

File: src/polyhash/index.js [hunk 3]
```diff
@@ -300,18 +384,22 @@ function _toTurfPolygon(inputPolygon) {
...
 /**
- * Convert the given polygon into a cluster of locationIds
+ * Convert a geometry into a cluster of locationIds
  *
- * @param {*} polygon, Array of coordinates or GeoJSON polygon
+ * @param {*} input, coordindates, locationIds, polygon or stringId
  * @param {*} locationIdPrecision
  */
-function toCluster(inputPolygon, locationIdPrecision) {
+function toCluster(input, locationIdPrecision) {
```
Comment (jake-unl) [inline src/polyhash/index.js:392]: Is there a reasonable default value we could assign to locationIdPrecision?
Then we could add this to a list of possible options to handle on clustering, e.g.
  minimiseLocationIds (default: true)
  breakdownEdges (default: false)

Names and default values can also be changed. What do you think @eamonnmcevoy ?

File: src/polyhash/index.js [hunk 4]
```diff
@@ -505,9 +593,10 @@ function _getIntesection(polygon1, polygon2) {
 module.exports = {
   toPolyhash,
   toCoordinates,
-  compressPolyhash,
-  decompressPolyhash,
+  toLocationIds,
   toCluster,
+  compressPolyhash: compress,
```
Comment (jake-unl) [inline src/polyhash/index.js:598]: These can be renamed to compress & decompress

File: src/polyhash/index.js [hunk 5]
```diff
@@ -79,39 +79,122 @@ const decode = [
...
+ function toLocationIds(input, locationIdPrecision = 9) {
+  try {
+    let result = [];
+    // If it is a list of location ids
+    if (Array.isArray(input) && input.length &&
+    typeof input[0] === 'string' || input[0] instanceof String) {
+        result = input;
+    }
+    else {
+      const coords =  toCoordinates(input)
```
Comment (jake-unl) [inline src/polyhash/index.js:177]: minor: spacing

File: src/polyhash/index.js [hunk 6]
```diff
@@ -300,18 +384,22 @@ function _toTurfPolygon(inputPolygon) {
 
 
 /**
- * Convert the given polygon into a cluster of locationIds
+ * Convert a geometry into a cluster of locationIds
  *
- * @param {*} polygon, Array of coordinates or GeoJSON polygon
+ * @param {*} input, coordindates, locationIds, polygon or stringId
```
Comment (jake-unl) [inline src/polyhash/index.js:389]: minor: spelling of coordinates :)

File: src/polyhash/index.js (hunk 3 above)
Comment (eamonnmcevoy) [reply src/polyhash/index.js:392]: Good idea, I would suggest 9 as a default precision since that is the cell precision we show on the map.

Comment (wailo) [reply src/polyhash/index.js]: Updated

File: src/polyhash/index.js (hunk 6 above)
Comment (wailo) [reply src/polyhash/index.js]: fixed

File: src/polyhash/index.js (hunk 2 above)
Comment (wailo) [reply src/polyhash/index.js]: fixed

File: src/polyhash/index.js (hunk 5 above)
Comment (wailo) [reply src/polyhash/index.js]: fixed

File: src/polyhash/index.js (hunk 1 above)
Comment (wailo) [reply src/polyhash/index.js]: Done

**Review by jake-unl — APPROVED** [review]

**Pull request merged**